src/BTrees/Interfaces.py

```diff
@@ -13,35 +13,28 @@
 ##############################################################################
 
 from zope.interface import Interface, Attribute
+from zope.interface.common.collections import ISized
+from zope.interface.common.sequence import IMinimalSequence
+from zope.interface.common.collections import IMapping
 
+# pylint:disable=inherit-non-class,no-method-argument,no-self-argument
+# pylint:disable=unexpected-special-method-signature
 
 class ICollection(Interface):
+    """
+    A collection of zero or more objects.
+
+    In a boolean context, objects implementing this interface are
+    `True` if the collection is non-empty, and `False` if the
+    collection is empty.
+    """
 
     def clear():
         """Remove all of the items from the collection."""
 
-    def __nonzero__():
-        """Check if the collection is non-empty.
-
-        Return a true value if the collection is non-empty and a
-        false value otherwise.
-        """
-
-
-class IReadSequence(Interface):
-
-    def __getitem__(index):
-        """Return the value at the given index.
-
-        An :class:`IndexError` is raised if the index cannot be found.
-        """
-
-    def __getslice__(index1, index2):
-        """Return a subsequence from the original sequence.
 
-        The subsequence includes the items from index1 up to, but not
-        including, index2.
-        """
+# Backwards compatibility alias. To be removed in 5.0
+IReadSequence = IMinimalSequence
```

Comment:
Should the scheduled removal use a deprecation warning to inform users to update their code?

Reply:
Great question!

In theory, yes, it probably should. That would require adding a dependency on `zope.deprecation`, which this package currently does not have. This package has the extremely minimal requirements of just `persistent` and `zope.interface`, and it seems desirable to me to keep the requirements minimal as this is such a low-level package.

So, I'm hoping we can get away with just docs-deprecation. From a practical/pragmatic standpoint, since nothing actually implemented any of these interfaces, there would be little incentive to use them. Searching GitHub finds essentially no references to "BTrees.Interfaces" or "IReadSequence" outside of copies of this package (there's a handful of references no the the `BTreesConflictError` defined here, but that seems to be it).

What do you think?

Reply:
Thank you for the research on the usage of these interfaces. I am okay with deprecating only in the documentation.